src/async-assignment.js

```diff
@@ -1,49 +1,106 @@
 const assignmentId = Number(urlPath.split('/')[2]);
+const assignmentName = document.querySelector("h2.page-title").innerHTML;
+
 const submitBtn = `a[href="/assignment/${assignmentId}/dropbox/submit"].link-btn`;
 const grade = '.grading-grade';
 const receivedGrade = '.grading-grade span.received-grade';
 
+let cancelInterval = false;
+let newBtn;
+
 // jQuery Document Ready
 $(function () {
-  if ($(submitBtn)[0] === undefined)
-    return console.error(
-      'One or more variables is undefined... Report to developer.'
-    );
 
-  setTimeout(checkCompleted, 500);
+    setTimeout(checkCompleted, 500);
+    $(submitBtn).parent().after('<div id="schoology-ext-md" style="margin-top: 10px;"></div>');
+    newBtn = $("#schoology-ext-md");
 });
 
 const checkCompleted = async () => {
-  await getStorage();
-  if (
-    $(submitBtn).text() !== 'Submit Assignment' &&
-    $(submitBtn).text() !== 'Edit Draft'
-  ) {
-    console.log('Assignment submitted! Clearing interval.');
-    return writeStorage();
-  }
-
-  if ($(receivedGrade)[0]) {
-    console.log('Assignment has recieved Grade... Clearing interval');
-    return writeStorage();
-  }
-
-  console.log('Assignment not submitted...');
-
-  setTimeout(checkCompleted, 500);
+    await getStorage();
+
+    let newBtn = $("#schoology-ext-md");
+
+    function getDesc() {
+        let toReturn = "error";
+        $.each(schoologyData, function(i, v) {
+            if (v.id === assignmentId) {
+                toReturn = v.desc;
+                return false;
+            }
+        });
+        return toReturn;
+    }
```

Comment:
```suggestion
     const getDesc = () => {
       for (assignment of schoologyData) {
         if (assignment.id !== assignmentId) return;
         return assignment.desc
       }
    }
```
I feel like this implementation would be a lot cleaner.